docs/content/patterns/crud-data-tables/index.mdx

```diff
@@ -0,0 +1,448 @@
+
+<ImageWithBorder
+	alt="Screenshot of an open side drawer with the title ‘Upload file’ and an empty state short form with text input, radio buttons, text input, a Select, File upload and form buttons ‘Upload file’ and ‘Cancel’"
+	src="/img/patterns/crud-tables-upload-undefined-list.png"
+/>
+
+### Validation and focus
+
+When a file is added in a subpage or drawer, return the user to the parent form page, set their focus on a Section alert to inform them if the upload was successful or an error has occurred.
+
+The uploaded file should also be visible in a table or list on the parent page. Provide an edit or remove action in the table row or list to enable files to be changed.
+
+```jsx live
+<Stack gap={3}>
+	<Stack gap={1.5}>
+		<H1>Upload undefined list of files to table (repeatable)</H1>
+	</Stack>
+	<SectionAlert
+		onClose={console.log}
+		title="FileName1.png was uploaded"
+		tone="successHigh"
+	/>
+	<Stack gap={2}>
+		<H2>List of equipment</H2>
+		<TableWrapper>
+			<Table>
+				<TableHead>
+					<TableRow>
+						<TableHeader scope="col">Item</TableHeader>
+						<TableHeader scope="col">Qty</TableHeader>
+						<TableHeader scope="col">Files</TableHeader>
+						<TableHeader scope="col">Actions</TableHeader>
+					</TableRow>
+				</TableHead>
+				<TableBody>
+					<TableRow>
+						<TableCell>Type 1</TableCell>
+						<TableCell>1</TableCell>
+						<TableCell as="th" scope="row">
+							FileName1.png
+						</TableCell>
+						<TableCell>
+							<TextLink aria-describedby="FileName1.png" href="#">
+								Remove
+							</TextLink>
+						</TableCell>
+					</TableRow>
+				</TableBody>
+			</Table>
+		</TableWrapper>
+		<ButtonGroup>
+			<Button variant="secondary" size="sm" iconBefore={PlusIcon}>
+				Upload file
+			</Button>
+		</ButtonGroup>
+	</Stack>
+</Stack>
+```
+
+If a required file is not uploaded before the page is submitted, show an error in the row where the file is missing and provide a section alert above the table to assist with error correction.
+
+## Upload a defined list of files to a data table
+
+A specific or defined list of files can be uploaded several ways. The number of items on the list and number of files for each list item will determine the appropriate pattern to use.
+
+The upload defined list of files pattern reduces the amount of form content on the parent page making it easier for users to complete the form, and keeps the users context and focus when providing all required metadata in the side drawer or subpage.
+
+### Empty state
+
+Before any files are uploaded provide an empty state. We prefer using a standard empty state rather than an empty table because the experience of the empty state will be better for screen reader users. They will not need to navigate through a table to find out it is empty.
```

Comment:
```suggestion
Before any files are uploaded, provide an empty state. Use a standard empty state rather than an empty table because the experience of the empty state is simpler for many users. Screen reader users will not need to navigate through a table to find out it is empty.
```